### PR TITLE
refactoring generate_cf

### PR DESCRIPTION
**Pull request by Grahams-Uncle**

### CODE REVIEW

File: nbs/05_methods.base.ipynb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
```
Comment (BirkhoffG) [inline nbs/05_methods.base.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #20.            x: jnp.ndarray, # Input to be explained
</pre><p>annotate as <code>x: Array</code> from now on </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (BirkhoffG) [inline nbs/05_methods.base.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #27.        @deprecated(removed_in='0.1.4', deprecated_in='0.1.4')
</pre><p>Don't specify <code>removed_in='0.1.4'</code> here </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: nbs/06_evaluate.ipynb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
```
Comment (BirkhoffG) [inline nbs/06_evaluate.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #26.        strategy = StrategyFactory.get_strategy(strategy)
</pre><p>Let's move this line in front of <code>current_time = time.time()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

File: nbs/methods/04_counternet.ipynb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
```
Comment (BirkhoffG) [inline nbs/methods/04_counternet.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    assert x.shape == cf.shape
</pre><p>Add another test case: <code>assert cf.shape == (29,)</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

File: nbs/methods/05_sphere.ipynb [hunk 4]
```diff
@@ -229,19 +229,18 @@
     "        if configs is None:\n",
```
Comment (BirkhoffG) [inline nbs/methods/05_sphere.ipynb:229]: <p>Can you check whether <code>GrowingSphere</code> generates different <code>cfs</code> or different <code>X</code>  ? Add a test case.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

File: nbs/methods/06_cchvae.ipynb [hunk 5]
```diff
@@ -21,7 +21,16 @@
    "cell_type": "code",
```
Comment (BirkhoffG) [inline nbs/methods/06_cchvae.ipynb:21]: <pre class="ql-syntax" spellcheck="false">Line #40.            rng = random.PRNGKey(0)
</pre><p>Specify seed via <code>CCHVAEConfigs</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: nbs/methods/07_vaecf.ipynb [hunk 6]
```diff
@@ -21,7 +21,16 @@
    "cell_type": "code",
```
Comment (BirkhoffG) [inline nbs/methods/07_vaecf.ipynb:21]: <pre class="ql-syntax" spellcheck="false">Line #35.        @auto_reshaping('x')
</pre><p>We should jit this function. Write <code>@partial(jax.jit, static_argnums=[0, 2])</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

File: nbs/methods/08_clue.ipynb [hunk 7]
```diff
@@ -21,7 +21,16 @@
    "cell_type": "code",
```
Comment (BirkhoffG) [inline nbs/methods/08_clue.ipynb:21]: <pre class="ql-syntax" spellcheck="false">Line #11.            self.rng_key = random.PRNGKey(0)
</pre><p>Again, specify the seed via configs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/ReLax/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>
